src/git.py

```diff
@@ -128,11 +128,7 @@ def _set_trailers(self, patch: Patch) -> str:
           return self._git.commit('--amend', '-F', f.name)
 
     def push_changes(self) -> str:
-        try:
-            return self._git.push(f'HEAD:refs/for/{self._branch}%notify=NONE')
-        except subprocess.CalledProcessError as e:
-            logging.warning('Failed to push upstream because %s.', e.output)
-            raise
+        pass
```

Comment:
Hmm, I would prefer this `DryRunGerritGit` to only contain decls + pass

This probably warrants another commit here first to trim down the interface, e.g. rename `def push_patch()` to `def _push_patch()`

I think the only thing we want to expose is apply_patchsetup_and_cleanup()?

Then the def would become

```python
class DryRunGerritGit:
     def apply_patchsetup_and_cleanup(...):
          pass
```

Reply:
I've gone ahead and trimmed the interface down as 5adaf15faf15a33982e915f2cd0b292edd35faec.
So you should just be able to rebase on master.

Might have some merge conflicts (but it felt like those would be unavoidable, even if you made the change yourself)

Reply:
I thought about that, but I kind of like it still trying to apply patches locally and doing some other stuff. I found it to provide some useful feedback.